service/storage_service.cc

```diff
@@ -6177,6 +6177,143 @@ static bool increases_replicas_per_rack(const locator::topology& topology, const
     return m[dst_rack] + 1 > max;
 }
 
+future<service::group0_guard> storage_service::get_guard_for_tablet_update() {
+    auto guard = co_await _group0->client().start_operation(_group0_as, raft_timeout{});
+    co_return guard;
+}
+
+future<bool> storage_service::exec_tablet_update(service::group0_guard guard, std::vector<canonical_mutation> updates, sstring reason) {
+    rtlogger.info("{}", reason);
+    rtlogger.trace("do update {} reason {}", updates, reason);
+    updates.emplace_back(topology_mutation_builder(guard.write_timestamp()).build());
```

Comment:
Better to bump up topology version, since task info is part of token metadata structure, in case we cache token metadata by version somewhere.

```
            .set_version(_topology_state_machine._topology.version + 1)
```

Reply:
I guess you meant tablet_task_info is part of token metadata because tablet_task_info is part of tmap and tmap is part of token metadata. 

Do we update _topology_state_machine._topology.version whenever tmap is changed? I understand why you want to update the _topology for the cache reasons. But this sounds weird because we do not really change any of the topology.